content/_code-samples/airgapped-wallet/py/airgapped-wallet.py

```diff
@@ -0,0 +1,200 @@
+    image = Image.open(get_path("/Wallet/transactionID.png"))
+    image.show()
+
+
+def get_path(file):
+    """
+    Get path (filesystem management)
+    """
+
+    global File_
+    # Checks what OS is being us
+    OS = platform.system()
+    usr = Path.home()
+
+    # Get PATH format based on the OS
+    if OS == "Windows":
+        File_ = PureWindowsPath(str(usr) + file)
+    if OS == "Linux":
+        File_ = PurePath(str(usr) + file)
+
+    return str(File_)
+
+
+def main():
+    global File, Path_
+    
+    # Gets the machine's operating system (OS)
+    OS = platform.system()
+    usr = Path.home()
+    if OS == "Windows":
+        # If it's Windows, use this path:
+        File = PureWindowsPath(str(usr) + '/Wallet')
+        Path_ = str(PureWindowsPath(str(usr)))
+    if OS == "Linux":
+        # If it's Linux, use this path:
+        File = PurePath(str(usr) + '/Wallet')
+        Path_ = str(PurePath(str(usr)))
+        
+    # If the Wallet's folder already exists, continue on
+    if os.path.exists(File):
+        while True:
+            ask = int(input("\n 1. Transact XRP"
+                            "\n 2. Generate an XRP wallet (read only)"
+                            "\n 3. Showcase XRP Wallet Address (QR Code)"
+                            "\n\n Enter Index: "
+                            ))
+
+            if ask == 1:
+                password = str(input("             Enter Password: "))
+                amount = float(input("\n           Enter XRP To Send: "))
+                destination = input("\n        Enter Destination: ")
+                wallet_sequence = int(input("\n    Enter Wallet Sequence: "))
+                ledger_sequence = int(input("\n    Enter Ledger Sequence: "))
+
+                sign_transaction(_xrp_amount=amount,
+                                 _destination=destination,
+                                 _ledger_seq=ledger_sequence,
+                                 _wallet_seq=wallet_sequence,
+                                 password=password
+                                 )
+
+                del destination, amount, wallet_sequence, ledger_sequence
+
+            if ask == 2:
+                _pub, _priv, _seed = create_wallet()
+
+            if ask == 3:
+                with open(get_path("/Wallet/public.txt"), "r") as f:
+                    print(f"\n  Wallet Address: {f.read()}")
+
+                image = Image.open(get_path("/Wallet/public.png"))
+                image.show()
+
+    else:
+        # If the Wallet's folder does not exist, create one and store wallet data (encrypted private key, encrypted seed, account address) 
+        pub, priv, seed = create_wallet()
+        
+        os.makedirs(File)
+        
+        img = qrcode.make(pub)
+        img.save(get_path("/Wallet/public.png"))
+        openimg = Image.open(get_path("/Wallet/public.png"))
+        
+        openimg.show()
+
+        password = str(input("\n        Enter Password: "))
```

Comment:
Just running the code I'm not sure where I can find that - Do I need to run the 'Create Wallet' step first - then fund that, then enter that secret here?

Reply:
(Entering dummy values results in a crash because `private.txt` does not exist)

Reply:
```
python airgapped-wallet.py
/Users/jacksonmills/Wallet

 1. Transact XRP
 2. Generate an XRP wallet (read only)
 3. Showcase XRP Wallet Address (QR Code)

 Enter Index: 2


       XRP WALLET CREDENTIALS
  Wallet Address: rpRaYChQJFNwcPFvQ7hgE52Bn8pRJxGQW2
            Seed: sEdSuKDGQbhgxL98UpZrrdDDNepT57D

 1. Transact XRP
 2. Generate an XRP wallet (read only)
 3. Showcase XRP Wallet Address (QR Code)

 Enter Index: 1
             Enter Password: sEdSuKDGQbhgxL98UpZrrdDDNepT57D

           Enter XRP To Send: 1

        Enter Destination: rpRaYChQJFNwcPFvQ7hgE52Bn8pRJxGQW2

    Enter Wallet Sequence: 3

    Enter Ledger Sequence: 20
Traceback (most recent call last):
  File "/Users/jacksonmills/Documents/wojake-xrpl-dev-portal/xrpl-dev-portal/content/_code-samples/airgapped-wallet/py/airgapped-wallet.py", line 201, in <module>
    main()
  File "/Users/jacksonmills/Documents/wojake-xrpl-dev-portal/xrpl-dev-portal/content/_code-samples/airgapped-wallet/py/airgapped-wallet.py", line 137, in main
    sign_transaction(_xrp_amount=amount,
  File "/Users/jacksonmills/Documents/wojake-xrpl-dev-portal/xrpl-dev-portal/content/_code-samples/airgapped-wallet/py/airgapped-wallet.py", line 39, in sign_transaction
    with open(get_path("/Wallet/private.txt"), "r") as f:
FileNotFoundError: [Errno 2] No such file or directory: '/Users/jacksonmills/Wallet/private.txt'
```

Reply:
Ah I understand now - This happened because I didn't complete the initial setup properly - I was alt-tabbed by the QR code appearing and I guess I didn't finish setting a password somehow. But there was no error which showed that until much later.

Reply:
(I have a Wallet folder and a public.png, but no password file)

Reply:
You need to run `airgapped-wallet.py` first to set the wallet up. Set your password and scan the QR to obtain the address, then you'd fund the account via a faucet.

I can see where/how this can go wrong, fixing. Good catch (yet again!)